File: src/test/java/uk/gov/justice/hmpps/prison/util/MoneySupportTest.java
package uk.gov.justice.hmpps.prison.util;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

import java.math.BigDecimal;
import java.math.RoundingMode;

public class MoneySupportTest {

    @Test
    public void When_AmountIs10_Then_PenceValueIs1000() {
        assertThat(MoneySupport.poundsToPence(new BigDecimal("10.00"))).isEqualTo(1000);
    }

    @Test
    public void When_AmountIs10_20_Then_PenceValueIs1020() {
        assertThat(MoneySupport.poundsToPence(new BigDecimal("10.20"))).isEqualTo(1020);
    }

    @Test
    public void When_AmountIs20pOnly_Then_PenceValueIs20() {
        assertThat(MoneySupport.poundsToPence(new BigDecimal("0.20"))).isEqualTo(20);
    }

    @Test
    public void When_PenceIs1000_Then_PoundValueIs10_00() {
        assertThat(MoneySupport.penceToPounds(1000L))
                .isEqualTo(new BigDecimal("10.00")
                        .setScale(2, RoundingMode.HALF_UP));
    }

    @Test
    public void When_PenceIs1020_Then_PoundValueIs10_20() {
        assertThat(MoneySupport.penceToPounds(1020L))
                .isEqualTo(new BigDecimal("10.20")
                        .setScale(2, RoundingMode.HALF_UP));
    }

}

File: src/test/java/uk/gov/justice/hmpps/prison/util/MoneySupportTest.kt
package uk.gov.justice.hmpps.prison.util

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.math.BigDecimal

class MoneySupportTest {
  @Test
  fun `when amount is 10 then pence value is 1000`() {
    assertThat(MoneySupport.poundsToPence(BigDecimal("10.00"))).isEqualTo(1000)
  }

  @Test
  fun `when amount is 10_20 then pence value is 1020`() {
    assertThat(MoneySupport.poundsToPence(BigDecimal("10.20"))).isEqualTo(1020)
  }

  @Test
  fun `when amount is 20p only then pence value is 20`() {
    assertThat(MoneySupport.poundsToPence(BigDecimal("0.20"))).isEqualTo(20)
  }

  @Test
  fun `when pence is 1000 then pound value is 10_00`() {
    assertThat(MoneySupport.penceToPounds(1000L)).isEqualByComparingTo("10.00")
  }

  @Test
  fun `when pence is 1020 then pounds value is 10_20`() {
    assertThat(MoneySupport.penceToPounds(1020L)).isEqualByComparingTo("10.20")
  }
}
